src/Microsoft.AspNetCore.Mvc.Core/Routing/ActionConstraintMatcherPolicy.cs

```diff
@@ -0,0 +1,237 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.ActionConstraints;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Matchers;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Mvc.Routing
+{
+    // This is a bridge that allows us to execute IActionConstraint instance when
+    // used with Matcher.
+    internal class ActionConstraintMatcherPolicy : MatcherPolicy, IEndpointSelectorPolicy
+    {
+        private static readonly IReadOnlyList<Endpoint> EmptyEndpoints = Array.Empty<Endpoint>();
+
+        // We need to be able to run IActionConstraints on Endpoints that aren't associated
+        // with an action. This is a sentinel value we use when the endpoint isn't from MVC.
+        internal static readonly ActionDescriptor NonAction = new ActionDescriptor();
+        
+        private readonly ActionConstraintCache _actionConstraintCache;
+
+        public ActionConstraintMatcherPolicy(ActionConstraintCache actionConstraintCache)
+        {
+            _actionConstraintCache = actionConstraintCache;
+        }
+
+        // Run really late.
+        public override int Order => 100000;
```

Comment:
So late! Is there some way to test that it is the latest?

Reply:
I don't really intend for it to be always always always last, I just don't want it to come before anything I can think of. There isn't too much science involed.